dotnet-desktop-guide/net/winforms/controls/how-to-dock-and-anchor.md

```diff
@@ -0,0 +1,118 @@
+description: Learn how to s
+ms.date: 05/25/2021
+dev_langs:
+  - "csharp"
+  - "vb"
+helpviewer_keywords:
+  - "Anchor property [Windows Forms], enabling resizable forms"
+  - "Windows Forms controls, screen resolutions"
+  - "resizing forms [Windows Forms]"
+  - "Windows Forms controls, size"
+  - "screen resolution and control display"
+  - "controls [Windows Forms], anchoring"
+  - "forms [Windows Forms], resizing"
+  - "Windows Forms, resizing"
+  - "controls [Windows Forms], positioning"
+  - "controls [Windows Forms], docking"
+  - "Explorer-style applications [Windows Forms], creating"
+  - "Windows Forms controls, filling client area"
+---
+# How to dock and anchor controls (Windows Forms .NET)
+
+If you're designing a form that the user can resize at run time, the controls on your form should resize and reposition properly. Controls have two properties that help with automatic placement and sizing, when the form changes size.
+
+- <xref:System.Windows.Forms.Control.Dock%2A?displayProperty=nameWithType>
+
+  Controls that are docked fill the edges of the control's container, either the form or a container control. For example, Windows Explorer docks its <xref:System.Windows.Forms.TreeView> control to the left side of the window and its <xref:System.Windows.Forms.ListView> control to the right side of the window. The docking mode can be any side of the control's container, or set to fill the remaining space of the container.
+
+  :::image type="content" source="./media/how-to-dock-and-anchor/dock-modes.png" alt-text="A windows form demonstrating the different dock modes for a control":::
+
+  Controls are docked in reverse z-order and the <xref:System.Windows.Forms.Control.Dock%2A> property interacts with the <xref:System.Windows.Forms.Control.AutoSize%2A> property. For more information, see [Automatic sizing](layout.md#automatic-sizing).
+
+- <xref:System.Windows.Forms.Control.Anchor%2A?displayProperty=nameWithType>
+
+  When an anchored control's form is resized, the control maintains the distance between the control and the anchor positions. For example, if you have a <xref:System.Windows.Forms.TextBox> control that is anchored to the left, right, and bottom edges of the form, as the form is resized, the <xref:System.Windows.Forms.TextBox> control resizes horizontally so that it maintains the same distance from the right and left sides of the form. The control also positions itself vertically so that its location is always the same distance from the bottom edge of the form. If a control isn't anchored and the form is resized, the position of the control relative to the edges of the form is changed.
+
+  :::image type="content" source="./media/how-to-dock-and-anchor/anchor-resize.gif" alt-text="A Windows form demonstrating the different anchor modes for a control":::
+
+For more information, see [Position and layout of controls](layout.md).
+
+[!INCLUDE [desktop guide under construction](../../includes/desktop-guide-preview-note.md)]
```

Comment:
This is far down the page for this alert box. Is it actually necessary? If it's intended as a warning that the documentation is incomplete, it might be worth stating that more directly, as otherwise it's not obvious what's meant by "under construction".